finality-aleph/src/network/manager/compatibility.rs

```diff
@@ -77,12 +80,10 @@ impl<M: Multiaddress> Decode for VersionedAuthentication<M> {
         match version {
             1 => Ok(V1(DiscoveryMessage::decode(input)?)),
             _ => {
-                let mut payload = vec![
-                    0;
-                    num_bytes
-                        .try_into()
-                        .map_err(|_| "input too big to decode")?
-                ];
+                if num_bytes > MAX_AUTHENTICATION_SIZE {
+                    Err("Authentication too big to decode. Is over 16 KiB")?;
```

Comment:
Perhaps "Authentication has unknown version and is encoded as more than 16KiB." would be better?